pkg/controller/elasticsearch/nodespec/readiness_probe.go

```diff
@@ -28,7 +28,25 @@ func NewReadinessProbe() *corev1.Probe {
 
 const ReadinessProbeScriptConfigKey = "readiness-probe-script.sh"
 const ReadinessProbeScript = `#!/usr/bin/env bash
-# Consider a node to be healthy if it responds to a simple GET on "/_cat/nodes?local"
+
+# fail should be called as a last resort to help the user to understand why the probe failed
+function fail {
+  timestamp=$(date --iso-8601=seconds)
+  echo "{\"timestamp\": \"${timestamp}\", \"message\": \"readiness probe failed\", "$1"}" | tee /proc/1/fd/2
+  exit 1
+}
+
+labels="` + volume.DownwardAPIMountPath + "/" + volume.LabelsFile + `"
+
+if [[ ! -f "${labels}" ]]; then
+  fail "\"reason\": \"${labels} does not exist\""
+fi
+
+# get Elasticsearch version from the downward API
+version=$(grep "elasticsearch.k8s.elastic.co/version" ${labels} | cut -d '=' -f 2)
```

Comment:
Nit: could be safer to use the `VersionLabelName` constant:

https://github.com/elastic/cloud-on-k8s/blob/d212d415a0651066ad2f6fd506929c0e3fa976e8/pkg/controller/elasticsearch/label/label.go#L27